### PR TITLE
Fix of getentropy() usage (OpenSSL_1_1_1-stable)

### PR DESCRIPTION
**Pull request by millikk**
**Brief**
Several embedded toolchains may provide dummy implemented `getentropy()` function which always returns -1 and sets errno to the `ENOSYS`. As a result the function `SSL_CTX_new()` fails to create a new context

More details: [Issue#13002: Getting random data failed on some embedded Linux platforms](https://github.com/openssl/openssl/issues/13002)

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
This is a PR against 1.1.1.  Master probably won't be a clean merge because the file has moved.

**Comment by t8m** [comment]
I am not sure this is trivial enough for CLA: trivial. Also could you please also submit a PR against master?

**Comment by millikk** [comment]
@t8m, My sorry about CLA, I just followed the contributors guide. [PR to master ](https://github.com/openssl/openssl/pull/13114) has also been created.

**Comment by paulidale** [comment]
Thanks for the master PR.

**Comment by paulidale** [comment]
I've added a CLA hold until we figure out things out.
Without the comment, I'd call this trivial.  With it is pushing things as @t8m mentioned.

**Comment by mattcaswell** [comment]
Not trivial IMO. Please can you submit a CLA?

https://www.openssl.org/policies/cla.html

**Comment by millikk** [comment]
So, I should agree with `Individual Contributor License Agreement` and send it by email, shouldn't I ? Also could I omit the `Mailing Address` and `Telephone` fields?

**Comment by t8m** [comment]
> So, I should agree with `Individual Contributor License Agreement` and send it by email, shouldn't I ? Also could I omit the `Mailing Address` and `Telephone` fields?

The physical address cannot be omitted. Telephone I do not know but I suppose it can. @mattcaswell ?

**Comment by mattcaswell** [comment]
>  Also could I omit the Mailing Address and Telephone fields?

Mailing Address is a mandatory field so should not be omitted. Telephone is preferred but we will accept a CLA without it.

**Comment by millikk** [comment]
I asked about `Mailing Address` because I'm worried about the privacy of my personal data. Well, I'll submit the CLA. But firstly I need some time to read the CLA carefully.

**Comment by richsalz** [comment]
It seems doubtful that if you submit a bogus address, anyone would ever find out.

**Comment by mspncp** [comment]
> I asked about `Mailing Address` because I'm worried about the privacy of my personal data. Well, I'll submit the CLA. But firstly I need some time to read the CLA carefully.

@millikk the essence of the CLA is that you are confirming that the change is genuinely your work, (i.e.,  you own the copyright, if there is any, of the contribution) and that you grant the OpenSSL project the license to use it. You are not giving up any ownership of your intellectual property. (Not that this is my viewpoint as a juristic layman, I am not a lawyer.)

The CLA is important for us, because it enables us to prove (if necessary) that we have the permission to license the contributed code under the apache license. Your personal data will be treated confidential and remain within the OpenSSL software foundation and not be used for any other purpose than the one mentioned above. In particular it will not be forwarded to anyone else.

So there is no need to follow @richsalz's bogus advice to provide a bogus address. ;-)

**Comment by millikk** [comment]
Thank you guys for the information. I submitted the CLA and updated the description in the commit

**Comment by mspncp** [comment]
> I submitted the CLA and updated the description in the commit

Fine, thank you very much. In this case the 'CLA:trivial' line can be removed from the  commit message. In fact, it would be helpful if you could remove it right away, because then @mattcaswell can just reopen this pull request when your CLA has been filed and we can see immediately whether everything went ok, i.e., whether the Bot accepts your license or complains.

**Review by mspncp — APPROVED** [review]
LGTM.

_(Note to @millikk: This pull request has now reached the necessary number of approvals. As soon as the CLA has been filed and the 24 hour grace period has expired, it pull request will be merged.)_

**Review by mspncp — APPROVED** [review]
LGTM

The 'CLA: trivial' still needs to be removed from the commit message, but this could also be done by the team member who merges the pull request.

_(Ups. This was intended to be the approval for the other pull request. :-) )_

**Comment by millikk** [comment]
@mspncp, So I just need to wait for the merge? :)

**Comment by mspncp** [comment]
Exactly. 

> @mspncp, So I just need to wait for the merge? :)

Exactly. Just sit back and enjoy 😉.

**Comment by mspncp** [comment]
Merged to master as 8e596a93bc266259f1ef0d56601e58bbfe18317a. Thank you @millikk and congratulations to your first contribution to OpenSSL!

**Comment by mspncp** [comment]
The last [force-push](https://github.com/openssl/openssl/compare/d82f8358ac142b86807499748f4ad3f8ad6bc3bb..c29e716b02711c152be5474db66570e2147158d8) only amended the commit message to remove the `CLA: trivial` label. No re-review required.

**Comment by mspncp** [comment]
Merged to 1.1.1 as b316d06dd99ec55f0a5275ca8ca4188848227b3b, thanks again!

**Comment by millikk** [comment]
Thank you all so much :)

**Pull request closed without merging**
